src/Kiota.Builder/Writers/RelativeImportManager.cs

```diff
@@ -34,29 +37,36 @@ public RelativeImportManager(string namespacePrefix, char namespaceSeparator)
                 _ => codeUsing.Declaration.TypeDefinition.Name.ToFirstCharacterUpperCase(),
             };
 
-            if(typeDef == null)
+            if (typeDef == null)
                 return (importSymbol, codeUsing.Alias, "./"); // it's relative to the folder, with no declaration (default failsafe)
-            else {
-                var importPath = GetImportRelativePathFromNamespaces(currentNamespace, 
+            else
+            {
+                var importPath = GetImportRelativePathFromNamespaces(currentNamespace,
                                                         typeDef.GetImmediateParentOfType<CodeNamespace>());
-                if(string.IsNullOrEmpty(importPath))
-                    importPath+= codeUsing.Name;
-                else
-                    importPath+= codeUsing.Declaration.Name.ToFirstCharacterLowerCase();
+                if (importPath == "./")
```

Comment:
I believe this still needs to be updated given our last discussion. Plus this is going to derail Go and Ruby. It needs to be moved in the typescript special manager